march_gait_selection/test/nosetests/test_gait_selection.py

```diff
@@ -1,78 +1,89 @@
+        self.gait_selection = deepcopy(self._gait_selection)
+        self.gait_version_map = deepcopy(self._gait_selection.gait_version_map)
+
+    # __init__ tests
+    def test_init_with_wrong_package(self):
+        with self.assertRaises(PackageNotFoundError):
+            GaitSelection('wrong', VALID_DIRECTORY)
+
+    def test_init_with_wrong_directory(self):
+        with self.assertRaises(FileNotFoundError):
+            GaitSelection(VALID_PACKAGE, 'wrong')
+
+    # gait version map setter tests
+    def test_set_gait_version_map_with_wrong_type(self):
+        with self.assertRaises(TypeError):
+            self.gait_selection.gait_version_map = 'wrong'
+
+    def test_set_gait_version_map_with_empty_version_map(self):
+        self.gait_version_map = {}
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    def test_set_gait_version_map_with_non_existing_gait_name(self):
+        self.gait_version_map['wrong'] = self.gait_version_map['walk']
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    def test_set_gait_version_map_with_non_existing_subgait_name(self):
+        self.gait_version_map['walk']['wrong'] = self.gait_version_map['walk']['right_open']
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    def test_set_gait_version_map_with_wrong_subgait_version(self):
+        self.gait_version_map['walk']['right_open'] = 'wrong'
+        with self.assertRaises(GaitError):
+            self.gait_selection.gait_version_map = self.gait_version_map
+
+    # load gaits tests
+    def test_types_in_loaded_gaits(self):
+        for gait in self.gait_selection.loaded_gaits:
+            self.assertIsInstance(gait, (Gait, BalanceGait))
+
+    # scan directory tests
+    def test_scan_directory_top_level_length(self):
+        directory = self.gait_selection.scan_directory()
+        self.assertTrue(len(['stairs_up', 'walk', 'walk_medium', 'walk_small']), len(directory.keys()))
+
+    def test_scan_directory_top_level_content(self):
+        directory = self.gait_selection.scan_directory()
+        self.assertTrue(all(key in ['stairs_up', 'walk', 'walk_medium', 'walk_small'] for key in directory.keys()))
+
+    def test_scan_directory_subgait_versions(self):
+        directory = self.gait_selection.scan_directory()
+        walk_subgaits = directory['walk']['subgaits']
+        self.assertEqual(walk_subgaits['left_swing'], ['MV_walk_leftswing_v2'])
+
+    # get item tests
+    def test_get_item_with_wrong_name(self):
+        gait = self.gait_selection['wrong']
+        self.assertEqual(gait, None)
+
+    def test_get_item_valid_gait_name(self):
+        self.assertFalse(self.gait_selection['walk'] is None)
+
+    def test_get_item_type(self):
+        gait = self.gait_selection['walk']
+        self.assertIsInstance(gait, Gait)
```

Comment:
```suggestion
        self.assertIsInstance(self.gait_selection['walk'], Gait)
```
(doing this makes the error message a bit more informative)